negotiation/src/main/java/it/eng/negotiation/service/ContractNegotiationConsumerService.java

```diff
@@ -154,44 +156,36 @@ public void handleAgreement(ContractAgreementMessage contractAgreementMessage) {
     /**
      * The response body is not specified and clients are not required to process it.
      *
-     * @param consumerPid
      * @param contractNegotiationEventMessage
-     * @return
      */
 
-    public JsonNode handleEventsResponse(String consumerPid, ContractNegotiationEventMessage contractNegotiationEventMessage) {
+    public void handleFinalizeEvent(ContractNegotiationEventMessage contractNegotiationEventMessage) {
+    	if (!contractNegotiationEventMessage.getEventType().equals(ContractNegotiationEventType.FINALIZED)) {
```

Comment:
Can we use here something like ContractNegotiationState.STATE_X.nextState()
Here and in all other places where we check if current state is correct

Reply:
I did not know that we have that, it will be changed right away 😄